### PR TITLE
Xiaoyu - Make moving summary to another week reset the summary checkmark

### PR DESCRIPTION
**Pull request by 20chen-7**
# Description

1. (PRIORITY MEDIUM) Jae: Make moving summary to another week reset the summary checkmark

a. If someone submitted their summary late for the week, and then forgot to submit it for the previous week and instead submitted it for this week, then it gets logged for this week and that (correctly) gives the green summary submitted checkmark. In cases like this, someone will sometimes need to move that summary to the last week. Right now, the green checkmark stays. If a summary is moved from this week to any other week, it should reset the checkmark. 

b. What we need is a "Move" button. This button will give the option to move the summary to "last week" or "week before last" or "choose date" and then open the submit modal with the information that is being moved.
Moving like this should clear the summary field for the current week and reset the green check so it’s clear the person still needs to submit a (new) summary for this week.

[Here's a video](https://www.dropbox.com/s/gu9m0mk9s2pqrdi/Move%20Summary%20Bug.mov?dl=0)

## Main changes explained:
- Add 'weekSelect' and creat radio including four options( 'this week', 'last week', Week Before Last', 'Three Weeks Ago'
- Create 'handleMoveOptions' function to reset the selected removed content and switch to weekSelect week.
- Disabled current week options in the current tab.

## How to test:
1. check into current branch
2. do `npm install` and `...` to run this PR locally
3. log as admin user
4. go to dashboard→ 'Weekly Summary Due Date(click to open)→ select the rest three options, and check the content and green checkmark after submiting the summary.

## Screenshots or videos of changes:

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/102696148/bdba5509-7dd6-47af-b6fc-22e9e8ef05d1

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/102696148/7c4a487a-9d21-40e3-a8e0-d86b0664f105

## Note:
After submitting the summary, you may require to refresh the website to see the green checkmark change.

### CODE REVIEW

**Review by leonzh2k — CHANGES_REQUESTED** [review]
seems to work:

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/49804462/a4c7011a-3699-4b6a-b647-f393c00738c3

From playing around with it, I think it's a bit unintuitive to use. Initially when I clicked a radio button, I thought it deleted my summary for that week, and had no idea that it moved the text to another week. Maybe it has to do with the wording "I confirm this summary is for..." Maybe it could be something like "Move this summary to..." and let the user explicitly know that checking a radio button will move their summary to the associated week

**Comment by AuroraLovesWhales** [comment]
Hi @20chen-7. I have tested your PR and the basic functionality works as intended. If I submit a summary for "Last Week/Week Before Last/Three Weeks Ago", the green checkmark does not appear. And if I submit a summary for "This Week" and refresh, the green checkmark appears:
https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/44281916/f2c5b873-1c8d-4842-bafc-2f1762ce8be2

However, I noticed an inconsistency between the weeks selected from the tab bar and the radio buttons. For example, if I click on "Week Before Last" from the tab bar, the same selection would be automatically updated for the radio button (greyed out). However, I am still able to click on "Last Week" from the radio button, save, and my summary would be saved for "Week Before Last":
https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/44281916/515dcbc5-6ae6-4971-bd8d-d20d44f164b2

This may cause some confusion to the user. Maybe there is a way to improve so that, when selecting a radio button for certain week, the tab bar would be automatically updated to that week as well (same way as how the radio button is updated when certain tab is selected in the current design), so that the two selection fields remain consistent both way.

File: src/components/WeeklySummary/WeeklySummary.jsx [hunk 1]
```diff
@@ -510,6 +559,47 @@ export class WeeklySummary extends Component {
...
+                        label="Last week"
+                        value="2"
+                        onChange={this.handleMoveOptions}
+                        disabled={activeTab === '2'}
+                      />
+                      <CustomInput
+                        id="weekBeforeLast"
+                        name="weekSelect"
+                        type="radio"
+                        label="Week fore Last"
```
[the remark this answers is not in the document]
Comment (20chen-7) [reply src/components/WeeklySummary/WeeklySummary.jsx]: I have fixed the typo and capitalization. Thank you!

**Comment by 20chen-7** [comment]
> Hi @20chen-7. I have tested your PR and the basic functionality works as intended. If I submit a summary for "Last Week/Week Before Last/Three Weeks Ago", the green checkmark does not appear. And if I submit a summary for "This Week" and refresh, the green checkmark appears: https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/44281916/f2c5b873-1c8d-4842-bafc-2f1762ce8be2
> 
> However, I noticed an inconsistency between the weeks selected from the tab bar and the radio buttons. For example, if I click on "Week Before Last" from the tab bar, the same selection would be automatically updated for the radio button (greyed out). However, I am still able to click on "Last Week" from the radio button, save, and my summary would be saved for "Week Before Last": https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/44281916/515dcbc5-6ae6-4971-bd8d-d20d44f164b2
> 
> This may cause some confusion to the user. Maybe there is a way to improve so that, when selecting a radio button for certain week, the tab bar would be automatically updated to that week as well (same way as how the radio button is updated when certain tab is selected in the current design), so that the two selection fields remain consistent both way.

I agree with your suggestions. I

**Comment by 20chen-7** [comment]
> Hi @20chen-7. I have tested your PR and the basic functionality works as intended. If I submit a summary for "Last Week/Week Before Last/Three Weeks Ago", the green checkmark does not appear. And if I submit a summary for "This Week" and refresh, the green checkmark appears: https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/44281916/f2c5b873-1c8d-4842-bafc-2f1762ce8be2
> 
> However, I noticed an inconsistency between the weeks selected from the tab bar and the radio buttons. For example, if I click on "Week Before Last" from the tab bar, the same selection would be automatically updated for the radio button (greyed out). However, I am still able to click on "Last Week" from the radio button, save, and my summary would be saved for "Week Before Last": https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/44281916/515dcbc5-6ae6-4971-bd8d-d20d44f164b2
> 
> This may cause some confusion to the user. Maybe there is a way to improve so that, when selecting a radio button for a certain week, the tab bar would be automatically updated to that week as well (the same way as how the radio button is updated when a certain tab is selected in the current design), so that the two selection fields remain consistent both way.

I agree with you, the options in the different tabs may confuse users. But for this pr,  I want to make consistent changes to the "tab name", and these options are consistent with the 'Tab Name'.

**Review by OAkinbode — COMMENTED** [review]
Hello Chen, 

The PR seems to work as intended. Well done. I can write to a previous week:

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/104801652/6d4f2954-6d46-4130-9f6e-a037d0211121

Or move a comment made this week to a previous week and have the check mark reset:

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/104801652/c1312a3f-13de-4052-be74-95b0f0c1d607

**Review by bienzguoa — APPROVED** [review]
Good job! Works as intended.

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/92126559/7dd01c5c-5534-4ab3-b0a0-d85204c01138

**Review by eduhorta — COMMENTED** [review]
Hi @20chen-7!

This is a tricky feature. Not an easy task.

The feature is working, great job! 👏

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/105796041/eaefb7d6-4f52-44c1-bdbb-68345dcbd671

But as mentioned by @leonzh2k, it's a bit unintuitive.

Here are my observations to help implement it:
If the user is writing a summary for "This Week" and clicks on "Last Week", all the content is moved to "Last Week" tab and the active tab remains "This Week".

Then, let's say you click on "Last Week" tab and decide to move it to "Three Weeks ago" but you've missclicked on "Week Before Last" radio check and then click on "Three Weeks ago" radio check which was your original intent. The content remains on the first click which in this case is "Week Before Last". Then, if you click on "Week Before Last" tab and tries to move it to "Three Weeks ago", the radio will be checked for it already.

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/105796041/7f07d749-2233-42ac-b8c8-c2a9e640ca6a

What do you think about @AuroraLovesWhales idea? So when the user clicks on a radio check it also changes to active tab to it's value.

**Comment by 20chen-7** [comment]
> Hi @20chen-7!
> 
> This is a tricky feature. Not an easy task.
> 
> The feature is working, great job! 👏
> 
>  2023-06-15.10-11-22.mp4 
> But as mentioned by @leonzh2k, it's a bit unintuitive.
> 
> Here are my observations to help implement it: If the user is writing a summary for "This Week" and clicks on "Last Week", all the content is moved to "Last Week" tab and the active tab remains "This Week".
> 
> Then, let's say you click on "Last Week" tab and decide to move it to "Three Weeks ago" but you've missclicked on "Week Before Last" radio check and then click on "Three Weeks ago" radio check which was your original intent. The content remains on the first click which in this case is "Week Before Last". Then, if you click on "Week Before Last" tab and tries to move it to "Three Weeks ago", the radio will be checked for it already.
> 
>  2023-06-15.10-18-26.mp4 
> What do you think about @AuroraLovesWhales idea? So when the user clicks on a radio check it also changes to active tab to it's value.

It is good idea. I have linked the move options with the tab.

**Comment by one-community** [comment]
Please change this so that it is a move function, not another button to  choose when submitting the summary. Clicking "move to other week" should present the same options you have now.

**Comment by one-community** [comment]
It is rare someone needs to move a summary, and they are moving a summary they already submitted, so that’s what this button is for. They can already submit a summary for last week or the week before by just clicking those tabs, entering it, and saving.

**Comment by one-community** [comment]
It needs to be super intuitive. I want to be able to tell people to just "go in like you are submitting a new summary, but move it to last week"

**Pull request closed without merging**